### PR TITLE
Feature/extend options

### PR DESCRIPTION
**Pull request by roesslerj**

### CODE REVIEW

File: src/main/java/de/retest/web/selenium/AutocheckingRecheckDriver.java [hunk 1]
```diff
@@ -29,13 +29,10 @@ public AutocheckingRecheckDriver( final RemoteWebDriver wrapped ) {
 		options = RecheckOptions.builder().build();
 	}
 
-	// TODO Use RecheckWebOptions
-	public AutocheckingRecheckDriver( final RemoteWebDriver wrapped, final RecheckOptions options,
-			final AutocheckingCheckNamingStrategy namingStrategy ) {
+	public AutocheckingRecheckDriver( final RemoteWebDriver wrapped, final RecheckWebOptions options ) {
```
Comment (beatngu13) [inline src/main/java/de/retest/web/selenium/AutocheckingRecheckDriver.java]: 1. Now that there is `RecheckWebOptions`, shouldn't we use it everywhere in recheck-web? For example, also in the `RecheckWebImpl` ctor?
2. Note that the changes of this PR are essentially breaking public API as one cannot pass `RecheckOptions` anymore. Therefore, only deprecate the old ones?
3. You could simplify this class as follows:

```java
public class AutocheckingRecheckDriver extends UnbreakableDriver {

	private RecheckWebImpl re;
	private final RecheckWebOptions options;
	private final AutocheckingCheckNamingStrategy namingStrategy;

	public AutocheckingRecheckDriver( final RemoteWebDriver wrapped ) {
		this( wrapped, RecheckWebOptions.builder().build() );
	}

	public AutocheckingRecheckDriver( final RemoteWebDriver wrapped, final RecheckWebOptions options ) {
		super( wrapped );
		this.options = options;
		namingStrategy = options.getNamingStrategy();
	}

	// ...

}
```

File: src/main/java/de/retest/web/selenium/RecheckWebOptions.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package de.retest.web.selenium;
+
+import de.retest.recheck.RecheckOptions;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.experimental.SuperBuilder;
+
+@SuperBuilder
+@Getter
+public class RecheckWebOptions extends RecheckOptions {
```
Comment (beatngu13) [inline src/main/java/de/retest/web/selenium/RecheckWebOptions.java]: New public API, therefore I suggest to add some Javadoc to the class as well as the field?

File: src/main/java/de/retest/web/selenium/AutocheckingRecheckDriver.java (hunk 1 above)
Comment (roesslerj) [reply src/main/java/de/retest/web/selenium/AutocheckingRecheckDriver.java]: Done

File: src/main/java/de/retest/web/selenium/AutocheckingRecheckDriver.java [hunk 3]
```diff
@@ -24,18 +24,20 @@
...
 
 	public AutocheckingRecheckDriver( final RemoteWebDriver wrapped ) {
+		this( wrapped, RecheckWebOptions.builder().build() );
+	}
+
+	@Deprecated
+	public AutocheckingRecheckDriver( final RemoteWebDriver wrapped, final RecheckOptions options ) {
 		super( wrapped );
+		this.options = options;
 		namingStrategy = new CounterCheckNamingStrategy();
```
Comment (beatngu13) [inline src/main/java/de/retest/web/selenium/AutocheckingRecheckDriver.java]: Automatically respect the default defined in `RecheckWebOptions`?

```suggestion
		namingStrategy = RecheckWebOptions.builder().build().getNamingStrategy();
```

File: src/main/java/de/retest/web/selenium/RecheckDriver.java [hunk 4]
```diff
@@ -15,9 +13,8 @@ public RecheckDriver( final RemoteWebDriver wrapped ) {
 		super( wrapped );
 	}
 
-	public RecheckDriver( final RemoteWebDriver wrapped, final RecheckOptions options,
-			final AutocheckingCheckNamingStrategy namingStrategy ) {
-		super( wrapped, options, namingStrategy );
+	public RecheckDriver( final RemoteWebDriver wrapped, final RecheckWebOptions options ) {
```
Comment (beatngu13) [inline src/main/java/de/retest/web/selenium/RecheckDriver.java]: This is also a breaking change.

File: src/main/java/de/retest/web/selenium/AutocheckingRecheckDriver.java [hunk 5]
```diff
@@ -24,18 +24,20 @@
...
-		options = RecheckOptions.builder().build();
 	}
 
-	// TODO Use RecheckWebOptions
-	public AutocheckingRecheckDriver( final RemoteWebDriver wrapped, final RecheckOptions options,
-			final AutocheckingCheckNamingStrategy namingStrategy ) {
+	public AutocheckingRecheckDriver( final RemoteWebDriver wrapped, final RecheckWebOptions options ) {
```
Comment (beatngu13) [inline src/main/java/de/retest/web/selenium/AutocheckingRecheckDriver.java]: This is also a breaking change.

File: src/main/java/de/retest/web/selenium/AutocheckingRecheckDriver.java (hunk 3 above)
Comment (roesslerj) [reply src/main/java/de/retest/web/selenium/AutocheckingRecheckDriver.java]: Ok, done.

File: src/main/java/de/retest/web/selenium/RecheckDriver.java (hunk 4 above)
Comment (roesslerj) [reply src/main/java/de/retest/web/selenium/RecheckDriver.java]: Granted ... but one I'd accept as being in the grey-area...

File: src/main/java/de/retest/web/selenium/AutocheckingRecheckDriver.java (hunk 5 above)
Comment (roesslerj) [reply src/main/java/de/retest/web/selenium/AutocheckingRecheckDriver.java]: Same here...

**Pull request merged**